eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/ComplexSettingsPass.php

```diff
@@ -79,10 +79,12 @@ private function createFactoryDefinition( $argumentValue, $dynamicSettings )
             array( $argumentValue )
         );
 
-        $definition->setFactoryClass(
-            'eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Configuration\ComplexSettings\ComplexSettingValueFactory'
+        $definition->setFactory(
+            array(
+                'eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Configuration\ComplexSettings\ComplexSettingValueFactory',
+                'getArgumentValue'
+            )
```

Comment:
can do that. I try to avoid it for 5.3 backports, but here it in not an issue as it will only be backported to 5.4.